services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveBatchManager.java

```diff
@@ -61,10 +60,12 @@ public CompletableFuture<ReceiveMessageResponse> processRequest(ReceiveMessageRe
         return queueAttributesManager.getReceiveMessageTimeout(rq, config.messageMinWaitDuration()).thenCompose(waitTimeMs -> {
             CompletableFuture<ReceiveMessageResponse> receiveMessageFuture = new CompletableFuture<>();
             receiveQueueBuffer.receiveMessage(receiveMessageFuture, numMessages);
-            CompletableFuture<ReceiveMessageResponse> timeoutFuture = new CompletableFuture<>();
-            executor.schedule(() -> timeoutFuture.complete(ReceiveMessageResponse.builder().build()), waitTimeMs.toMillis(),
-                              TimeUnit.MILLISECONDS);
-            return receiveMessageFuture.applyToEither(timeoutFuture, Function.identity());
+            executor.schedule(() -> {
+                if (!receiveMessageFuture.isDone()) {
```

Comment:
I don't think we need to check isDone here, complete will handle it.

Reply:
If the user passes this value then user can control the Polling time out , this is behaviour same as https://github.com/aws/aws-sdk-java/blob/61d73631fac8535ad70666bbce9e70a1d2cea2ca/aws-java-sdk-sqs/src/main/java/com/amazonaws/services/sqs/buffered/ReceiveQueueBuffer.java#L192-L197

I removed this in earlier PR but when I was doing V1 comparison I realized that V1 uses this value to determine the Client side Wait time for that  particular request.